python/tskit/metadata.py

```diff
@@ -455,6 +455,10 @@ def enforce_fixed_properties(obj):
             elif type(obj) == dict:
                 ret = {k: enforce_fixed_properties(v) for k, v in obj.items()}
                 if ret.get("type") == "object":
+                    if ret.get("additional_properties") is True:
```

Comment:
Any reason for ``is True`` here? This is normally a no-no. Also, not clear why we're using ``ret.get()`` here rather than ``ret[]``.

Reply:
Just realised that additional_properties can be an object so this needs to change. Will fix.

Reply:
additional_properties is optional, hence the use of get. As it can be an object I've reverted to truthiness. Should be the right thing now.

Reply:
Hah, I didn't realise that the default default value for ``get`` was None!